StateT slides

jdesiloniz opened the pull request:
This PR contains the content of the slides for the StateT videos.

jdesiloniz commented:
@raulraja there are some code examples that don't carry any explanations in the slides as I plan to do that with doodles and the script. Also, I aknowledge that I don't follow exactly the structure of the previous videos as we've covered a lot of stuff already with the `State` video, even though I think most of the basics about `StateT` and monad transformers are explained here at some early level. Hope it's OK!

raulraja commented on StateT.md:
```diff
@@ -0,0 +1,271 @@
+- Slack : [https://kotlinlang.slack.com/messages/C5UPMM0A0](https://kotlinlang.slack.com/messages/C5UPMM0A0)
+- FP with Arrow 
+- 47 Degrees : [http://47deg.com](http://47deg.com)
+- @raulraja : [https://twitter.com/JaviTaiyou](https://twitter.com/JaviTaiyou)
```
s/@raulraja/@JaviTaiyou

raulraja commented:
@jdesiloniz sounds good, ping me when the script is ready for review and please use `^` deckset's speaker notes in this same document which is how we have been doing it lately and works out better for @benjymontoya and reviewers. Thanks!
https://docs.decksetapp.com/English.lproj/Presenting/01-presenter-notes.html

jdesiloniz commented:
@raulraja hey Raul, I've updated the slides adding the script under the speaker notes. I've also moved a lot of the text in the slides to be in the script so code is more readable. Hope it works OK!

raulraja commented on StateT.md:
```diff
@@ -0,0 +1,300 @@
+
+# StateT :: Map transformations
+
+We can also use `map` to define transformations on our values before updating our state:
```
`map` should come before `flatMap` and `binding`

raulraja commented on StateT.md:
````diff
@@ -0,0 +1,300 @@
+
+We can also use `map` to define transformations on our values before updating our state:
+
+```kotlin
````
Missing an example with Applicative `map` with multiple operations which is what would happen uf a user presses 4 buttons at the same time when getting into the elevator.
```kotlin
val context = StateT.monad<EitherPartialOf<ElevatorError>, Elevator>(Either.monad())
with (context) {
  map(op1, op2, op3, op4, { ... })
}
```

jdesiloniz commented:
@raulraja sorry for the delay. Could you take a look at the provided example for the Applicative builder before I tag Maureen for the language review? Thanks!

jdesiloniz commented:
@MaureenElsberry Hi! :). I understand that the contents of the slides/script won't experience further changes. Could you take a look at these? Thanks a lot!!

jdesiloniz commented:
@MaureenElsberry Thanks a lot!!

@raulraja if you don't have any concerns, I'd merge this to move on the recording of the script :)

JorgeCastilloPrz commented:
Hi! Isn't the script already there? I can see the ^comments below each slide.

jdesiloniz commented:
@JorgeCastilloPrz Awesome, thanks Jorge!! :)
